californium-core/src/main/java/org/eclipse/californium/core/network/TcpMatcher.java

```diff
@@ -211,16 +156,7 @@ public void completed(final Exchange exchange) {
 					exchangeStore.releaseToken(idByToken);
```

Comment:
if (!exchange.getCurrentRequest().getOptions().hasObserve()) {
					exchangeStore.releaseToken(idByToken);
				}

request with observe option 1 may be cleaned up on completion. rebase and use new Request.isObserve() ?